awsbestpractices/checking-object-presence/CheckingObjectPresence.java

```diff
@@ -0,0 +1,239 @@
+                    return null;
+                }
+
+                if ( !retryPolicy.allowRetry(retryCount++, System.currentTimeMillis() - startMs, RetryLoop.getDefaultRetrySleeper()) )
+                {
+                    exhibitor.getLog().add(ActivityLog.Type.ERROR, "Retries exhausted: " + ActivityLog.getExceptionMessage(e));
+                    return null;
+                }
+            }
+        }
+
+        final Throttle      throttle = makeThrottle(configValues);
+        final InputStream   in = object.getObjectContent();
+        final InputStream   wrappedstream = new InputStream()
+        {
+            @Override
+            public void close() throws IOException
+            {
+                in.close();
+            }
+
+            @Override
+            public int read() throws IOException
+            {
+                throttle.throttle(1);
+                return in.read();
+            }
+
+            @Override
+            public int read(byte[] b) throws IOException
+            {
+                int bytesRead = in.read(b);
+                if ( bytesRead > 0 )
+                {
+                    throttle.throttle(bytesRead);
+                }
+                return bytesRead;
+            }
+
+            @Override
+            public int read(byte[] b, int off, int len) throws IOException
+            {
+                int bytesRead = in.read(b, off, len);
+                if ( bytesRead > 0 )
+                {
+                    throttle.throttle(bytesRead);
+                }
+                return bytesRead;
+            }
+        };
+
+        return new BackupStream()
+        {
+            @Override
+            public InputStream getStream()
+            {
+                return wrappedstream;
+            }
+
+            @Override
+            public void close() throws IOException
+            {
+                in.close();
+            }
+        };
+    }
+
+    // used
+    // Simplified version of used_3
+    public BackupStream doesObjectExist_used_4() {
+        S3Object object = null;
+        while ( object == null ) {
+            try {
+                object = s3.getObject("foo", "bar");
+            }
+            catch ( AmazonS3Exception e) {
+                return null;
+            }
+        }
+
+        System.out.println(object.getObjectContent());
+    }
+
+    // Sets field
+    public void doesObjectExist_setToField() {
+        try {
+            this.objectMetadata = s3.getObjectMetadata();
+        }
+        catch (Exception e) {
+        }
+        return null;
+    }
+
+    // Read to file
+    public void doesObjectExist_readToFile() {
+        try {
+            s3.getObject(new GetObjectRequest("foo", "bar"), new File("my-file-path"));
+        } catch (Exception e) {
+        }
+    }
+
+    // Can be replaced
+    public boolean doesBucketExist_headBucket() {
+        try {
+            s3.headBucket(new HeadBucketRequest());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is manually checking for the existence of a bucket in S3. You could use the SDK (operation name: [doesBucketExist](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#doesBucketExist-java.lang.String-) or [doesBucketExistV2](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#doesBucketExistV2-java.lang.String-)) to automatically check for this instead.